File: templates/problems/15_greedy/06_merge_triplets/merge_triplets_test.go
package merge_triplets

import "testing"

func TestMergeTriplets(t *testing.T) {
	tests := []struct {
		name     string
		triplets [][]int
		target   []int
		expected bool
	}{
		{"example true", [][]int{{2, 5, 3}, {1, 8, 4}, {1, 7, 5}}, []int{2, 7, 5}, true},
		{"example false", [][]int{{3, 4, 5}, {4, 5, 6}}, []int{3, 2, 5}, false},
		{"exact match single", [][]int{{2, 5, 3}}, []int{2, 5, 3}, true},
		{"no valid triplet", [][]int{{1, 1, 1}}, []int{2, 2, 2}, false},
		{"filter out exceeding", [][]int{{2, 5, 3}, {10, 1, 1}, {1, 7, 5}}, []int{2, 7, 5}, true},
		{"all exceed one dim", [][]int{{3, 1, 1}, {3, 2, 2}}, []int{2, 2, 2}, false},
		{"multiple combos", [][]int{{1, 2, 3}, {2, 1, 3}, {2, 2, 1}}, []int{2, 2, 3}, true},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := mergeTriplets(tt.triplets, tt.target)
			if got != tt.expected {
				t.Errorf("mergeTriplets(%v, %v) = %v, want %v", tt.triplets, tt.target, got, tt.expected)
			}
		})
	}
}

File: templates/problems/15_greedy/06_merge_triplets/typescript/mergeTriplets.test.ts
import { describe, it, expect } from 'vitest';
import { mergeTriplets } from './mergeTriplets';

describe('MergeTriplets', () => {
    it('example true', () => {
        expect(mergeTriplets([[2, 5, 3], [1, 8, 4], [1, 7, 5]], [2, 7, 5])).toBe(true);
    });

    it('example false', () => {
        expect(mergeTriplets([[3, 4, 5], [4, 5, 6]], [3, 2, 5])).toBe(false);
    });

    it('exact match single', () => {
        expect(mergeTriplets([[2, 5, 3]], [2, 5, 3])).toBe(true);
    });

    it('no valid triplet', () => {
        expect(mergeTriplets([[1, 1, 1]], [2, 2, 2])).toBe(false);
    });

    it('filter out exceeding', () => {
        expect(mergeTriplets([[2, 5, 3], [10, 1, 1], [1, 7, 5]], [2, 7, 5])).toBe(true);
    });

    it('all exceed one dim', () => {
        expect(mergeTriplets([[3, 1, 1], [3, 2, 2]], [2, 2, 2])).toBe(false);
    });

    it('multiple combos', () => {
        expect(mergeTriplets([[1, 2, 3], [2, 1, 3], [2, 2, 1]], [2, 2, 3])).toBe(true);
    });
});
